### PR TITLE
chore: prepare `4.0.0b1` release

### PR DESCRIPTION
**Pull request by vytas7**
I have now rendered out the Towncrier newsfragments, and `git`-removed them.

### CODE REVIEW

File: docs/changes/4.0.0.rst [hunk 1]
```diff
@@ -4,17 +4,30 @@ Changelog for Falcon 4.0.0
...
 
-Although we are still in the process of polishing the last items from the
-`version 4.0 milestone <https://github.com/falconry/falcon/milestone/34>`__,
-we do not expect (m)any radical changes from now on until Falcon 4.0.0 final.
-The full changelog of new features, breaking changes and fixes will be rendered
-as part of the first beta release.
+If you make use of type annotations in your Falcon app, please run your type
+checker of choice without any *typeshed* extensions for Falcon, and
+:ref:`report back to us <chat>`.
+Our ambition is to support ``mypy --strict`` or equivalent out of the box!
```
Comment (CaselIT) [inline docs/changes/4.0.0.rst:17]: I don't think we do at the moment. Or at list I don't think I have ever tried.

File: docs/changes/4.0.0.rst [hunk 2]
```diff
@@ -4,17 +4,30 @@ Changelog for Falcon 4.0.0
...
+Please also check the list of **breaking changes** below.
 
-Although we are still in the process of polishing the last items from the
-`version 4.0 milestone <https://github.com/falconry/falcon/milestone/34>`__,
-we do not expect (m)any radical changes from now on until Falcon 4.0.0 final.
-The full changelog of new features, breaking changes and fixes will be rendered
-as part of the first beta release.
+If you make use of type annotations in your Falcon app, please run your type
+checker of choice without any *typeshed* extensions for Falcon, and
+:ref:`report back to us <chat>`.
```
Comment (CaselIT) [inline docs/changes/4.0.0.rst:16]: maybe add "if you find any issues"?

**Review by CaselIT — COMMENTED** [review]
Just left a couple of comments, but it otherwise looks nice!

File: docs/changes/4.0.0.rst (hunk 1 above)
Comment (CaselIT) [reply docs/changes/4.0.0.rst:17]: > Found 154 errors in 29 files (checked 65 source files)

At a quick range they seem doable. I could try squeezing a PR in the beta phase

Comment (vytas7) [reply docs/changes/4.0.0.rst:17]: I meant running on users' code, not Falcon itself. Or was that your app?

I know that E2E tests server app passes with strict.

File: docs/changes/4.0.0.rst (hunk 2 above)
Comment (vytas7) [reply docs/changes/4.0.0.rst:16]: I think we want to hear success stories too!

Comment (CaselIT) [reply docs/changes/4.0.0.rst:16]: ok!

File: docs/changes/4.0.0.rst (hunk 1 above)
Comment (CaselIT) [reply docs/changes/4.0.0.rst:17]: oh it wasn't very clear

Comment (vytas7) [reply docs/changes/4.0.0.rst:17]: I think from the user's perspective it should be clear, but I am open to removing that statement altogether.

Comment (vytas7) [reply docs/changes/4.0.0.rst]: Removed this piece of marketing for now.

**Review by CaselIT — APPROVED** [review]

**Pull request merged**